src/vulkan/engine_vulkan_debugger.cc

```diff
@@ -0,0 +1,976 @@
+// Copyright 2020 The Amber Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "src/vulkan/engine_vulkan.h"
+
+#if AMBER_ENABLE_VK_DEBUGGING
```

Comment:
I don't think this is needed. We should just not add this file in the cmakelist instead of if'defing the whole file.

Reply:
There are two functions that need a definition, which are stubbed at the bottom of this file:

```C++
std::pair<Engine::Debugger*, Result> EngineVulkan::GetDebugger() {
  return {nullptr,
          Result("Amber was not built with AMBER_ENABLE_VK_DEBUGGING enabled")};
}

void EngineVulkan::DeleteDebugger() {}
```

You want me to move their definition to `engine_vulkan.cc`?

I personally prefer to keep this sort of thing in the preprocessor if possible as it doesn't add additional complexity to each build system. I see that amber has CMake and Android build scripts. Anything file-conditional like this would need to be replicated carefully in both (plus any other build systems that might crop up in the future).

Reply:
The problem with reviewing lineally is you forget to remove comments when you realize they don't make sense. This bit is fine to do this way since we have a #else block later. If it was just conditionally blocking out all the code I'd say do it in the cmake file.